client/app/core/dialog-field-refresh.service.js

```diff
@@ -60,7 +60,7 @@ export function DialogFieldRefreshFactory(CollectionsApi, EventNotifications) {
         dialogField.default_value = new Date(newDialogField.values);
       } else {
         if (angular.isUndefined(newDialogField.default_value) || newDialogField.default_value === null
-          || newDialogField.default_value === '') {
+          || newDialogField.default_value === '' || dialogField.type === 'DialogFieldCheckBox') {
```

Comment:
Only thing I would suggest is pulling out this long `if` statement into a separate "private" method with a better name around what this is doing. `noDefaultValueOrCheckbox`? Struggling to think of a great name, maybe we could just take out the first part so it would be something like `blankDefaultValue() || dialogField.type === 'DialogFieldCheckBox'`